xdem/coreg.py

```diff
@@ -1664,55 +1715,154 @@ def _fit_func(self, ref_dem: np.ndarray, tba_dem: np.ndarray, transform: Optiona
         self._meta["bias"] = bias
         self._meta["resolution"] = resolution
 
-    def _apply_func(self, dem: np.ndarray, transform: rio.transform.Affine) -> np.ndarray:
-        """Apply the estimated x/y/z offsets to a DEM."""
-        bounds, resolution = _transform_to_bounds_and_res(dem.shape, transform)
-        scaling_factor = self._meta["resolution"] / resolution
+    def _to_matrix_func(self) -> np.ndarray:
+        """Return a transformation matrix from the estimated offsets."""
+        offset_east = self._meta["offset_east_px"] * self._meta["resolution"]
+        offset_north = self._meta["offset_north_px"] * self._meta["resolution"]
 
-        # Make index grids for the east and north dimensions
-        east_grid = np.arange(dem.shape[1]) * scaling_factor
-        north_grid = np.arange(dem.shape[0]) * scaling_factor
+        matrix = np.diag(np.ones(4, dtype=float))
+        matrix[0, 3] += offset_east
+        matrix[1, 3] += offset_north
+        matrix[2, 3] += self._meta["bias"]
 
-        # Make a function to estimate the DEM (used to construct an offset DEM)
-        elevation_function = scipy.interpolate.RectBivariateSpline(x=north_grid, y=east_grid,
-                                                                   z=np.where(np.isnan(dem), -9999, dem))
-        # Make a function to estimate nodata gaps in the aligned DEM (used to fix the estimated offset DEM)
-        nodata_function = scipy.interpolate.RectBivariateSpline(x=north_grid, y=east_grid, z=np.isnan(dem))
+        return matrix
 
-        shifted_east_grid = east_grid + self._meta["offset_east_px"]
-        shifted_north_grid = north_grid - self._meta["offset_north_px"]
 
-        shifted_dem = elevation_function(y=shifted_east_grid, x=shifted_north_grid)
-        new_nans = nodata_function(y=shifted_east_grid, x=shifted_north_grid)
-        shifted_dem[new_nans >= 1] = np.nan
+def invert_matrix(matrix: np.ndarray) -> np.ndarray:
+    """Invert a transformation matrix."""
+    with warnings.catch_warnings():
+        # Deprecation warning from pytransform3d. Let's hope that is fixed in the near future.
+        warnings.filterwarnings("ignore", message="`np.float` is a deprecated alias for the builtin `float`")
 
-        shifted_dem += self._meta["bias"]
+        # Invert the transform if wanted.
+        return pytransform3d.transformations.invert_transform(matrix)
```

Comment:
Does it differ from numpy.linalg.inv?

Reply:
Seems like it's the same! I'll change to `np.linalg.inv`
```python
In [1]: import numpy as np

In [2]: import pytransform3d.transformations

In [3]: m = pytransform3d.transformations.random_transform()
In [4]: m
Out[4]: 
array([[-0.69885409, -0.18464974, -0.69101913,  1.45427351],
       [-0.57588809,  0.71823881,  0.39049446,  0.76103773],
       [ 0.42421206,  0.67084834, -0.60828171,  0.12167502],
       [ 0.        ,  0.        ,  0.        ,  1.        ]])

In [5]: assert np.array_equal(pytransform3d.transformations.invert_transform(m), np.linalg.inv(m))
```

Reply:
Changed the invert_matrix to be a bit more useful in 5c5777e and also sanity-check the matrix. Therefore, pytransform3d is still necessary, and can be kept as is.